packages/analytics-adapter/src/adapter.test.ts

```diff
@@ -0,0 +1,426 @@
+import {BaseAdapter} from "./adapter";
+
+let logSpy: any;
+
+class ExampleAdapter extends BaseAdapter {
+    checkAnalyticsProviders(): void {
+    }
+
+    getAnalytics(): any {
+        // @ts-ignore
+        return window['example-analytics']
+    }
+
+    sendMetrics(type: string, event: any): void {
+        this.emit([type, event])
+    }
+}
+
+const firstCid = {
+    "cid": "cid1:eid1",
+    "eid": "eid1",
+    "uid": "user1",
+};
+
+const secondCid = {
+    "cid": "cid2:eid2",
+    "eid": "eid2",
+    "uid": "user1",
+};
+
+describe('Adapter Test', () => {
+    beforeEach(() => {
+        delete window['evolv'];
+        // @ts-ignore
+        delete window['example-analytics'];
+        logSpy = jest.spyOn(console, "log").mockImplementation();
+    });
+
+    afterEach(() => {
+        logSpy.mockClear();
+    });
+
+    describe('Handle missing Analytics ', () => {
+        test('Log message if Analytics is not found', (done) => {
+            window['evolv'] = {
+                client: {
+                    on: jest.fn()
+                }
+            };
+
+            const client = new ExampleAdapter(0);
+
+            setTimeout(() => {
+                expect(logSpy.mock.calls.length).toBe(1);
+                expect(logSpy.mock.calls[0][0]).toBe("Evolv: Analytics integration timed out - Couldn't find Analytics");
+                done();
+            }, client.interval);
+        });
+    });
+
+    describe('Ensure listeners set up', () => {
+        test('Validate on listener configured when evolv ready before constructor', () => {
+            var analytics = jest.fn();
```

Comment:
Nitpick: `const` or `let` instead of `var`.

Reply:
Done